tests/foreman/ui/test_hostcollection.py

```diff
@@ -383,3 +404,223 @@ def test_negative_hosts_limit(self):
+            'repository-set': REPOSET['rhst7'],
+            'repository': REPOS['rhst7']['name'],
+            'organization-id': self.org_.id,
+            'content-view-id': self.content_view.id,
+            'lifecycle-environment-id': self.env.id,
+            'activationkey-id': self.activation_key.id,
+        })
+        setup_org_for_a_custom_repo({
+            'url': FAKE_0_YUM_REPO,
+            'organization-id': self.org_.id,
+            'content-view-id': self.content_view.id,
+            'lifecycle-environment-id': self.env.id,
+            'activationkey-id': self.activation_key.id,
+        })
+        self.hosts = []
+        for _ in range(self.hosts_number):
+            client = VirtualMachine(distro='rhel71')
+            self.hosts.append(client)
+            client.create()
+            client.install_katello_ca()
+            result = client.register_contenthost(
+                self.org_.label, self.activation_key.name)
+            self.assertEqual(result.return_code, 0)
+            client.enable_repo(REPOS['rhst7']['id'])
+            client.install_katello_agent()
+        host_ids = [
+            entities.Host().search(query={
+                'search': 'name={0}'.format(host.hostname)})[0].id
+            for host in self.hosts
+        ]
+        self.host_collection = entities.HostCollection(
+            host=host_ids,
+            organization=self.org_,
+        ).create()
+
+    def tearDown(self):
+        """Destroy all the VMs"""
+        for client in self.hosts:
+            client.destroy()
+        super(HostCollectionPackageManagementTest, self).tearDown()
+
+    def _validate_package_installed(self, hosts, package_name,
+                                    expected_installed=True, timeout=120):
+        """Check whether package was installed on the list of hosts."""
+
+        for host in hosts:
+            for _ in range(timeout / 15):
+                result = self.contenthost.package_search(
+                    host.hostname, package_name)
+                if (result is not None and expected_installed or
+                        result is None and not expected_installed):
+                    break
+                sleep(15)
+            else:
+                self.fail(
+                    u'Package {0} was not {1} host {2}'.format(
+                        package_name,
+                        'installed on ' if expected_installed else
+                        'removed from ',
+                        host.hostname,
+                    )
+                )
+
+    @tier3
+    def test_positive_install_package(self):
+        """Install a package to hosts inside host collection remotely
+
+        @id: eead8392-0ffc-4062-b045-5d0252670775
+
+        @assert: Package was successfully installed on all the hosts in host
+        collection
+
+        @CaseLevel: System
+        """
+        with Session(self.browser) as session:
+            session.nav.go_to_select_org(self.org_.name)
```

Comment:
Awesome explanation, many thanks for checking that.
But basically this approach completely eliminates all the advantages setUpClass gives us, making no point in setUpClass and basically increasing time needed for tests execution a lot. @rochacbruno @elyezer do we really need that `--boxed` flag?